BrowserKit/Sources/SiteImageView/Entities/SiteImageModel.swift

```diff
@@ -4,51 +4,55 @@
     let id: UUID
 
-    /// The image type expected when making a request
-    let expectedImageType: SiteImageType
+    // The image type expected when making a request
+    let imageType: SiteImageType
 
-    /// Represents the website and not the resource
-    let siteURLString: String?
+    // Represents the website (not the associated image resource)
+    let siteURL: URL
 
-    /// URL can be nil in case the urlStringRequest cannot be used to build a URL
-    var siteURL: URL?
+    // Used to cache any resources related to this request
+    let cacheKey: String
 
-    /// Used to cache any resources related to this request
-    var cacheKey: String = ""
+    // URL of the image (e.g. faviconURL, preferrably high resolution)
+    var resourceURL: URL?
 
-    /// Domain can be nil in case we don't have a siteURL to get the domain from
-    var domain: ImageDomain?
-
-    /// The favicon URL scrapped from the webpage, high resolution found at preference
-    var faviconURL: URL?
-
-    /// The fetched or generated favicon image
-    public var faviconImage: UIImage?
-
-    /// The fetched hero image
-    var heroImage: UIImage?
+    // Loaded image asset
+    public var image: UIImage?
 
     public init(id: UUID,
-                expectedImageType: SiteImageType,
-                siteURLString: String? = nil,
-                siteURL: URL? = nil,
-                cacheKey: String = "",
-                faviconURL: URL? = nil,
-                faviconImage: UIImage? = nil,
-                heroImage: UIImage? = nil) {
+                imageType: SiteImageType,
+                siteURL: URL,
+                resourceURL: URL? = nil,
+                image: UIImage? = nil) {
         self.id = id
-        self.expectedImageType = expectedImageType
-        self.siteURLString = siteURLString
+        self.imageType = imageType
         self.siteURL = siteURL
-        self.cacheKey = cacheKey
-        self.domain = nil
-        self.faviconURL = faviconURL
-        self.faviconImage = faviconImage
-        self.heroImage = heroImage
+        self.cacheKey = SiteImageModel.generateCacheKey(siteURL: siteURL, type: imageType)
+        self.resourceURL = resourceURL
+        self.image = image
+    }
+
+    /// Generates a cache key for the given image type by using its associated site URL.
+    /// - Parameters:
+    ///   - siteURL: The website with which this image is associated.
+    ///   - type: The image type.
+    /// - Returns: A cache key value for storing this image in an image cache.
+    static func generateCacheKey(siteURL: URL, type: SiteImageType) -> String {
+        switch type {
+        case .heroImage:
+            // Always use the full site URL as the cache key for hero images
+            return siteURL.absoluteString
+        case .favicon:
+            // FIXME Why would we ever want to use a hard-coded faviconURL (sponsored tile, suggested tile, etc...)
+            // as the cache key for either a URL or Image?
```

Comment:
@OrlaM Do you know why we'd ever want to cache a URL and Image based solely on its exact faviconURL? I imagine we only know the exact URL for sponsored tiles and default suggested sites. Maybe pinned tiles. 

I don't see why we wouldn't just use the shortDomain and avoid making more requests for users who browse to those sites later.

FYI, here's the original code for generating the cache key.
```
    private func generateCacheKey(siteURL: URL?,
                                  faviconURL: URL? = nil,
                                  type: SiteImageType) -> String {
        // If we already have a favicon url use the url as the cache key
        if let faviconURL = faviconURL {
            return faviconURL.absoluteString
        }

        guard let siteURL = siteURL else { return "" }

        // Always use the full site URL as the cache key for hero images
        if type == .heroImage {
            return siteURL.absoluteString
        }

        // For everything else use the domain as the key to avoid caching
        // and fetching unnecessary duplicates
        return siteURL.shortDomain ?? siteURL.shortDisplayString
    }
```

Reply:
I don't recall the context but I think it was probably for the sponsored tile use case, in the case of a sponsored tile we really want to prefer to use the image provided. Another approach could be to still use the domain as the key but have some mechanism to force replace the image when we are providing one in cases like the sponsored tile?

Reply:
Hmm...! I was wondering about this too. I wouldn't expect it to be different but it's probably good to ensure we use that URL when it's hard coded _just_ in case.

Before the refactor, it was a little mysterious when you should call the original `generateCacheKey` method with or without the `faviconURL` param. Potential for getting yourself into trouble there. So I'll push up an extra check in `SiteImageModel`'s `init` to use the hard coded faviconURL when we have one (rather than update `generateCacheKey`).